MediaBrowser.Api/System/ActivityLogService.cs

```diff
@@ -56,7 +56,10 @@ public object Get(GetActivityLogs request)
                 DateTime.Parse(request.MinDate, null, DateTimeStyles.RoundtripKind).ToUniversalTime();
 
             var filterFunc = new Func<IQueryable<ActivityLog>, IQueryable<ActivityLog>>(
-                entries => entries.Where(entry => entry.DateCreated >= minDate));
+                entries => entries.Where(entry => entry.DateCreated >= minDate
+                                                  && (!request.HasUserId.HasValue || (request.HasUserId.Value
```

Comment:
I don't like the ternary. Maybe `request.HasUserId != true || entry.UserId != Guid.Empty` is better, shorter at least.

Reply:
The issue with the condition you suggested is that there are 3 states - HasUserId is null, HasUserId is true, HasUserId is false.
The filter must only be applied when HasUserId is true/false, and ignored when HasUserId is null.